bundles/http/Citrix/Microsoft-Outlook/script.js

```diff
@@ -0,0 +1,319 @@
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'email_address_name',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'type',
+						type: 'STRING',
+						length: 255
+					}
+				]
+			},
+			{
+				name: 'my_events',
+				columns: [
+					{
+						name: 'i_cal_u_id',
+						type: 'STRING',
+						length: 255,
+						primaryKey: true
+					},
+					{
+						name: 'id',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'recurrence_pattern_day_of',
+						type: 'INTEGER'
+					},
+					{
+						name: 'recurrence_pattern_type',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'recurrence_range_end_date',
+						type: 'DATE'
+					}
+				]
+			}
+		],
+		relationships: [
+			{
+				name: 'nested_table_1',
+				primaryTable: 'calender_view',
+				foreignTable: 'calender_view_attendees',
+				columnPairs: [
+					{
+						primaryKey: 'i_cal_u_id',
+						foreignKey: 'parent_i_cal_u_id'
+					}
+				]
+			},
+			{
+				name: 'fk_my_events_id',
+				primaryTable: 'calender_view',
+				foreignTable: 'my_events',
+				columnPairs: [
+					{
+						primaryKey: 'series_master_id',
+						foreignKey: 'id'
+					}
+				]
+			},
+			{
+				name: 'fk_calender_v_organizer_e',
+				primaryTable: 'users',
+				foreignTable: 'calender_view',
+				columnPairs: [
+					{
+						primaryKey: 'user_principal_name',
+						foreignKey: 'organizer_email_address_a'
+					}
+				]
+			}
+		]
+	}
+});
+
+const moment = library.load('moment-timezone');
+const uuid = library.load('uuid');
+const startDate = moment.utc().subtract(1, 'd').format();
+const endDate = moment.utc().add(30, 'd').format();
+const pageSize = 100;
+let userId = [];
+
+async function fullSync({ client, dataStore }) {
+	let URL = `/v1.0/users?$top=${pageSize}`;
+	let nextPage = '';
+	do {
+		const userRequest = await client.fetch(URL);
+		if (!userRequest.ok) {
+			throw new Error(`Users sync failed ${userRequest.status}:${userRequest.statusText}.`);
+		}
+		const userResponse = await userRequest.json();
+		for (const value of userResponse.value) {
+			userId.push(value.id);
+			dataStore.save('user', {
+				id: value.id,
+				mail: value.mail,
+				user_principal_name: value.userPrincipalName,
+				display_name: value.displayName
+			});
+		}
+		nextPage = userResponse['@odata.nextLink'] ?? null;
+		if (nextPage != null) URL = `/v1.0/users?${nextPage.split(`?`)[1]}`;
+	} while (nextPage);
+	await calendarView(client, dataStore, userId);
+	await myEvents(client, dataStore, userId);
+}
+
+async function calendarView(client, dataStore, userId) {
+	for (const id of userId) {
+		const calendarViewRequest = await client.fetch(
+			`/v1.0/users/${id}/calendarView?startdatetime=${startDate}&enddatetime=${endDate}&$top=${pageSize}`
+		);
+		if (!calendarViewRequest.ok && calendarViewRequest.status != 404) {
+			throw new Error(`Calendar sync failed ${calendarViewRequest.status}:${calendarViewRequest.statusText}.`);
+		}
```

Comment:
```suggestion
		if (!calendarViewRequest.ok ) {
		        if(calendarViewRequest.status != 404)
			   throw new Error(`Calendar sync failed ${calendarViewRequest.status}:${calendarViewRequest.statusText}.`);
			else
			   continue
		}
```
Lets handle all the error at one place, as for some errors like 500 there will be no json body so at line 264 it will throw an error.

Reply:
> if (!calendarViewRequest.ok && calendarViewRequest.status != 404) {
			throw new Error(`Calendar sync failed ${calendarViewRequest.status}:${calendarViewRequest.statusText}.`);
		}
		
> if (!calendarViewRequest.ok ) {
		        if(calendarViewRequest.status != 404)
			   throw new Error(`Calendar sync failed ${calendarViewRequest.status}:${calendarViewRequest.statusText}.`);
			else
			   continue
		}
		
I believe both does the same operation. Correct me if I'm wrong.